### PR TITLE
Fix changing envs

### PR DESCRIPTION
**Pull request by crpc**
+ other minor fixes

### CODE REVIEW

File: src/utils/helpers/GraphHelpers.ts [hunk 1]
```diff
@@ -334,10 +383,19 @@ export function findGroupBackgroundDimension(nodesGroup: DependencyNode[]) {
     let rightLimitNode = nodesGroup[0];
 
     nodesGroup.forEach((node: DependencyNode) => {
-        if (!node.x || !node.y || !rightLimitNode.x || !leftLimitNode.x || !upperLimitNode.y || !lowerLimitNode.y) {
+        if (
+            !node.x ||
```
Comment (Fiszcz) [inline src/utils/helpers/GraphHelpers.ts]: is it possible that fields from this ``if`` statement have values equal to ``0`` ? If yes, should we stop go further, to the following ifs ?

File: src/utils/helpers/GraphHelpers.ts [hunk 2]
```diff
@@ -75,6 +75,16 @@ export function highlight(clickedNode: DependencyNode) {
...
 }
 
+function getDefaultScaleValue(container: NodeSelection, dimension: ReturnType<typeof findGroupBackgroundDimension>) {
+    if (!dimension) {
+        return 1;
+    }
+    const width = Number(container.attr('width'));
+    const height = Number(container.attr('height'));
+
+    return Math.min(1.3, 0.9 / Math.max(dimension.width / width, dimension.height / height));
```
Comment (fernard) [inline src/utils/helpers/GraphHelpers.ts]: What is the reasoning behind this scale calculation? For me it's completely opaque.

File: src/utils/helpers/DetailsDrawHelpers.ts [hunk 3]
```diff
@@ -189,9 +194,15 @@ export function initializeDetailsView(node: TreeNode) {
     const height = Number(detailsViewContainer.attr('height'));
 
     const detailsZoom = selectDetailsZoom();
-
     createDiagrams(detailsZoom, consumerNodes, providerNodes, width, height);
     switchDetailsVisibility();
+
+    const detailsContainerDiv = selectDetailsContainerDiv().node();
+    const { width: containerWidth, height: containerHeight } = detailsContainerDiv
```
Comment (michaube) [inline src/utils/helpers/DetailsDrawHelpers.ts]: should't this logic be inside resetZoomPosition ?

File: src/utils/helpers/GraphHelpers.ts (hunk 1 above)
Comment (michaube) [reply src/utils/helpers/GraphHelpers.ts]: Should this node even exist with those values empty ? If it should maybe have a helper function for this check, like isNodeEmpty

File: src/utils/helpers/GraphHelpers.ts (hunk 2 above)
Comment (crpc) [reply src/utils/helpers/GraphHelpers.ts]: It chooses either biggest possible scale value(1.3 zoom value is a perfect to show focus on certain nodes and still allow user to see a bit wider context) or 90% of selected dimension to absolute(not dependent of zoom value) client width/height ratio - whatever is bigger.

I haven't thought about rewriting this before and I have no idea if the idea of deriving this value could be improved but I agree that it has some magic numbers and it isn't clear at all how the scale value is being derived.

I will try to extract some of the calculations to consts so it gets more readable

File: src/utils/helpers/DetailsDrawHelpers.ts (hunk 3 above)
Comment (crpc) [reply src/utils/helpers/DetailsDrawHelpers.ts]: It should 👍 , gonna change

File: src/utils/helpers/GraphHelpers.ts (hunk 1 above)
Comment (crpc) [reply src/utils/helpers/GraphHelpers.ts]: Do You agree on me fixing this logic in another PR? It looks like a bigger change than extracting some logic to a function

**Pull request merged**